examples/blaze/src/main/scala/com/example/http4s/blaze/demo/server/endpoints/TimeoutHttpEndpoint.scala

```diff
@@ -0,0 +1,17 @@
+package com.example.http4s.blaze.demo.server.endpoints
+
+import cats.effect.Sync
+import cats.syntax.flatMap._
+import org.http4s._
+import org.http4s.dsl.Http4sDsl
+
+import scala.util.Random
+
+class TimeoutHttpEndpoint[F[_]](implicit F: Sync[F]) extends Http4sDsl[F] {
+
+  val service: HttpService[F] = HttpService {
+    case GET -> Root / ApiVersion / "timeout" =>
+      F.delay(Thread.sleep(Random.nextInt(3) * 1000L)).flatMap(_ => Ok())
```

Comment:
I think we should use an fs2 scheduler here.  Teach people Thread.sleep is never the answer.

Reply:
+1 on the scheduler.

Reply:
I tried to make it work with Scheduler before but I remember something didn't work well that's why I used Thread.sleep but I'm aware of the misuse. So here I tried again an this is the result:

This is the shortest version that works for me (any less verbose version?):
```scala
val randomMillis = FiniteDuration(Random.nextInt(3) * 1000L, TimeUnit.MILLISECONDS)
val p = Scheduler(2).flatMap(_.delay(Stream.emit("").covary[F], randomMillis)).map(Ok(_))
p.compile.fold(NoContent())((_, y) => y).flatten
```

And I expected this one to work but it doesn't:

```scala
val randomMillis = FiniteDuration(Random.nextInt(3) * 1000L, TimeUnit.MILLISECONDS)
Ok(Scheduler(2).flatMap(_.delay(Stream.emit("").covary[F], randomMillis)))
```
Is this a bug or is it expected?

When I say it doesn't work I mean it doesn't give a timeout. It seems that the response body is made before passing through the Timeout Middleware.

Reply:
That's going to start a scheduler on every request.  I think passing in a bracketed scheduler would make for a better example.

Reply:
As ross said, that's literally creating a scheduler, which also creates a thread pool every time. 

Service can be redone to:

`def service(scheduler: Scheduler[F]): HttpService[F]`

Reply:
Good point. I'll rework this example :+1:

Reply:
I guess this one is more acceptable?

```scala
val randomMillis = FiniteDuration(Random.nextInt(3) * 1000L, TimeUnit.MILLISECONDS)
S.sleep(randomMillis).compile.drain *> Ok("delayed response")
```

Reply:
Compiles in my head:

```scala
S.effect.delay(Ok("delayed response"), randomMillis)
```

Since it's a `FiniteDuration`, it's misleading to call it `millis`.

Reply:
Awesome, I just learned something new :)  Didn't know about the `effect` method on Scheduler.

Reply:
```scala
F.delay(Random.nextInt(3)).flatMap { d =>
    S.effect.delay(Ok(), d.seconds)
}
```
Compiles in my head.
That being said, using `Stream` the way you did can be interrupted while sleeping (doesn't matter in this case though)